pkg/helmdeployer/deployer.go

```diff
@@ -352,6 +356,102 @@ func (h *helm) install(bundleID string, manifest *manifest.Manifest, chart *char
 	return u.Run(releaseName, chart, vals)
 }
 
+func (h *helm) getValues(options fleet.BundleDeploymentOptions, defaultNamespace string) (map[string]interface{}, error) {
+	if options.Helm == nil {
+		return nil, nil
+	}
+
+	var values map[string]interface{}
+	if options.Helm.Values != nil {
+		values = options.Helm.Values.Data
+	}
+
+	if h.secretCache == nil || h.configmapCache == nil {
+		return values, nil
+	}
+
+	for _, valuesFrom := range options.Helm.ValuesFrom {
+		var val map[string]interface{}
+		if valuesFrom.SecretKeyRef != nil {
+			name := valuesFrom.SecretKeyRef.Name
+			namespace := valuesFrom.SecretKeyRef.Namespace
+			if namespace == "" {
+				namespace = defaultNamespace
+			}
+			key := valuesFrom.SecretKeyRef.Key
+			if key == "" {
+				key = "values.yaml"
+			}
+			secret, err := h.secretCache.Get(namespace, name)
+			if err != nil {
+				return nil, err
+			}
+			data, ok := secret.Data[key]
+			if !ok {
+				return nil, fmt.Errorf("key %s is missing from secret %s/%s, can't use it in valuesFrom", key, namespace, name)
+			}
+			if err := yaml.Unmarshal(data, &val); err != nil {
+				return nil, err
+			}
+		} else if valuesFrom.ConfigMapKeyRef != nil {
```

Comment:
Don't we want to check both the Secret and the ConfigMap? Should this be a separate `if` block?

Reply:
The api is designed to only respect either secretRef or configmapRef. This followed the same valueFrom dedign from k8s pod api.